examples/persistent-counters/counter.go

```diff
@@ -29,21 +41,11 @@ func (pc *PersistentCounter) Get(key string) (int64, error) {
 }
 
 func (pc *PersistentCounter) Inc(key string, value int64) error {
-	presentVal, err := pc.Get(key)
-	if err != nil {
-		if err == badger.ErrKeyNotFound {
-			presentVal = 0
-		} else {
-			return err
-		}
-	}
-	err = pc.db.Update(func(txn *badger.Txn) error {
-		keyBytes := []byte(key)
-		valBytes := make([]byte, 64)
-		binary.PutVarint(valBytes, presentVal+value)
-		err = txn.Set(keyBytes, valBytes)
-		return err
-	})
+	mergeOp := pc.db.GetMergeOperator([]byte(key), counterMerge, time.Millisecond*1)
+	defer mergeOp.Stop()
+	incValBytes := make([]byte, 64)
```

Comment:
better to put these 2 lines before starting merge. Also instead of relying on defer, better to close it much earlier